pathlib: Fix glob() to return Path objects rather than strings.

RotatingLlama opened the pull request:
Fix glob() to return Path objects rather than bare strings, same as CPython https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob

RotatingLlama committed:
pathlib: Fix glob() to return Path objects rather than strings.

Fix glob() to return Path objects rather than bare strings, same as CPython
https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob

andrewleech commented:
This discrepancy with cpython caught me out recently too, thanks for submitting the fix. 

Would you be able to update the commit message to match the repo standard needs patterns and force-push to the same branch? That would fix the ci failure.

RotatingLlama commented:
No probs, it's very minor. This is actually my first pull request and I spent about half an hour yesterday trying to fix the error, but couldn't figure out how to do it. Would you give me a pointer, please?

RotatingLlama committed:
Merge branch 'micropython:master' into patch-1

RotatingLlama committed:
pathlib: Fix glob() to return Path objects rather than strings.
Fix glob() to return Path objects rather than bare strings, same as CPython
https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob

RotatingLlama commented:
I've now spent literal hours of my life trying to understand how to fix this pull request, and am still no further forward.  I'm not willing to waste any more time for the sake of such a minor correction.  I'm going to see if I can delete it for the sake of keeping the place tidy, but if it won't let me do that either then I'm out.

dpgeorge commented:
I redid this in #1120.

RotatingLlama commented:
Happy days!  Sorry I failed so utterly to get my own PR working 🤦
